rest/models/snapshot.go

```diff
@@ -69,6 +69,120 @@ func (p GetGainersLosersSnapshotParams) WithIncludeOTC(q bool) *GetGainersLosers
+	Sort  *Sort  `query:"sort"`
+	Order *Order `query:"order"`
+}
+
+// NewGetOptionsChainParams will return a pointer to GetOptionsChainParams.
+func NewGetOptionsChainParams(ticker string) *GetOptionsChainParams {
+	return &GetOptionsChainParams{Ticker: &ticker}
+}
+
+// WithTicker sets underlying ticker symbol to params
+func (o *GetOptionsChainParams) WithTicker(ticker string) *GetOptionsChainParams {
+	o.Ticker = &ticker
+	return o
+}
+
+// WithStrikePrice sets strike price to params. Strike Price is the price at which a put or call option can be exercised.
+func (o *GetOptionsChainParams) WithStrikePrice(strikePrice string) *GetOptionsChainParams {
+	o.StrikePrice = &strikePrice
+	return o
+}
+
+// WithContractType sets contract type to params.
+// contractType options include ContractTypeCall and ContractTypePut.
+func (o *GetOptionsChainParams) WithContractType(contractType ContractType) *GetOptionsChainParams {
+	o.ContractType = &contractType
+	return o
+}
+
+// WithLimit sets number of results returned. Limit default is 10. Limit must fall in range of 0-1000.
+func (o *GetOptionsChainParams) WithLimit(limit int) *GetOptionsChainParams {
+	o.Limit = &limit
+	return o
+}
+
+// WithExpirationDate sets expiration_date query parameter.
+// comparator options include EQ, LT, LTE, GT, and GTE.
+// expirationDate should be in YYYY-MM-DD format
+func (o *GetOptionsChainParams) WithExpirationDate(comparator Comparator, expirationDate string) *GetOptionsChainParams {
+	switch comparator {
+	case EQ:
+		o.ExpirationDateEQ = &expirationDate
+	case LT:
+		o.ExpirationDateLT = &expirationDate
+	case LTE:
+		o.ExpirationDateLTE = &expirationDate
+	case GT:
+		o.ExpirationDateGT = &expirationDate
+	case GTE:
+		o.ExpirationDateGTE = &expirationDate
+	default:
+		o.ExpirationDateEQ = &expirationDate
+	}
+	return o
+}
+
+// WithOrder sets order of results based on the Sort field.
+func (o *GetOptionsChainParams) WithOrder(order Order) *GetOptionsChainParams {
+	switch order {
+	case Asc:
+		o.Order = &order
+	case Desc:
+		o.Order = &order
+	}
+	return o
+}
+
+// WithSort sets sort field. Sort expects to receive TickerSymbol, ExpirationDate, or StrikePrice as an argument.
+func (o *GetOptionsChainParams) WithSort(sort Sort) *GetOptionsChainParams {
+	switch sort {
+	case TickerSymbol:
+		o.Sort = &sort
+	case ExpirationDate:
+		o.Sort = &sort
+	case StrikePrice:
+		o.Sort = &sort
+	}
+	return o
+}
+
+type GetOptionsChainSnapshotResponse struct {
+	BaseResponse
+	Results []OptionsChainSnapshot `json:"results,omitempty"`
+}
+
+type OptionsChainSnapshot struct {
+	Day               DaySnapshot     `json:"day,omitempty"`
+	Details           OptionDetails   `json:"details,omitempty"`
+	Greeks            Greeks          `json:"greeks,omitempty"`
+	LastQuote         LastQuote       `json:"last_quote,omitempty"`
+	UnderlyingAsset   UnderlyingAsset `json:"underlying_asset,omitempty"`
+	ImpliedVolatility *float64        `json:"implied_volatility,omitempty"`
```

Comment:
Response fields should never be pointers

Reply:
Removed this struct and replaced with the OptionsContractSnapshot struct that was already being used.